spacy/pipeline/sentencizer.pyx

```diff
@@ -152,6 +154,7 @@ class Sentencizer(Pipe):
         """
         cfg = srsly.msgpack_loads(bytes_data)
         self.punct_chars = set(cfg.get("punct_chars", self.default_punct_chars))
+        self.overwrite = cfg.get("overwrite", False)
```

Comment:
Do we need this default `False` value here? If the setting is not in `cfg`, we can just keep `self.overwrite` as it was by default after construction? Actually isn't the same true also for `self.punct_chars`?

So you'd have something like 
``` suggestion
        if "punct_chars" in cfg:
            self.punct_chars = set(cfg.get("punct_chars"))
        if "overwrite" in cfg:
            self.overwrite = cfg.get("overwrite", False)
```
and then the same in `from_disk` and potentially some other components.

I just want to be mindful of these hidden defaults creeping back into the code...

Reply:
I agree about hidden defaults, but I don't think this is a hidden default. This is the hard-coded setting from <3.2.0 which isn't going to change, so if you're loading a model that was serialized with v3.1 or earlier, you get the same default in cases where this `cfg` value is missing no matter what you've specified on init. If the `cfg` value is present, it's v3.2+ and you get the stored value. (If you're hacking your `cfg`, well, that's on you.)

I think something similar happened in the `punct_chars` development, but I'd have to check the history to be sure.

I would keep both of these as is.

Reply:
Ah, I see what you mean. Then I'd personally prefer putting this value in a var somewhere and document that it's for bwd-compat only.

Reply:
I think this is fine as-is? If the var is otherwise unused I think we can accomplish the same thing with a comment here.

Reply:
I think this is fine as-is. If the var isn't used for anything else then I think a comment explaining why this is the way it is will accomplish the same thing.

Reply:
The same value is used twice within `Sentencizer`'s IO methods, so let's just pull it out into a `default_` var and document the bwd-compat reasoning.